lib/browser/api/browser-window.js

```diff
@@ -26,6 +27,33 @@ BrowserWindow.prototype._init = function () {
     ipcMain.emit('ELECTRON_GUEST_WINDOW_MANAGER_WINDOW_OPEN', event, url, frameName, disposition, options)
   })
 
+  this.webContents.on('-web-contents-created', (event, webContents, url,
+                                                frameName) => {
+    pendingWebContents.set(webContents, {url, frameName})
+  })
+  // Create a new browser window for the native implementation of
+  // "window.open"(sandboxed mode only)
+  this.webContents.on('-add-new-contents', (event, webContents, disposition,
+                                            userGesture, left, top, width,
+                                            height) => {
+    if ((disposition !== 'foreground-tab' && disposition !== 'new-window') ||
+        !pendingWebContents.has(webContents)) {
+      return
+    }
+
+    let {url, frameName} = pendingWebContents.get(webContents)
+    pendingWebContents.delete(webContents)
+    const options = {
+      show: true,
+      x: left,
+      y: top,
+      width: width || 800,
+      height: height || 600,
+      webContents: webContents
+    }
+    ipcMain.emit('ELECTRON_GUEST_WINDOW_MANAGER_WINDOW_OPEN', event, url, frameName, disposition, options)
```

Comment:
I think we should let `NativeWindow` manage the `webContents`, letting `guest-window-manager` do the management makes the whole logic too complicated.

It can be done by overriding a method of `WebContentsObserver` in `NativeWindow`.